File: easy/two_sum.rb
# @param {Integer[]} nums
# @param {Integer} target
# @return {Integer[]}
def two_sum(nums, target)
  map = {}
  nums.each_with_index do |i_value, i_index|
    map[i_value] = i_index
  end
  # map = { 3: 0, 4: 1, 2: 2}

  nums.each_with_index do |i_value, i_index|
    if map[target - i_value] && map[target - i_value] != i_index
      return [map[target - i_value], i_index].sort
    end
  end
  return []
end

File: easy/two_sum.py
class Solution(object):
  def twoSum(self, nums, target):
    """
    :type nums: List[int]
    :type target: int
    :rtype: List[int]
    """
    map = {}
    for index in range(len(nums)):
      map[str(nums[index])] = index

    for index in range(len(nums)):
      if map.get(str(target - nums[index])) != None and map.get(str(target - nums[index])) != index:
        return sorted([map.get(str(target - nums[index])), index])
    return []

obj = Solution()
print(obj.twoSum([3,2,4], 6))
